BUG: DataFrame.round() drops column index name #11986

rcarneva opened the pull request:
Fixes #11986. Used the same reconstruction method as `DataFrame.isin()` to ensure that column names are kept through rounding. Works with `np.round` as well.

Pretty minor fix -- does this need a release note or anything else?

rcarneva committed:
BUG: DataFrame.round() drops column index name

jreback commented:
tests!

rcarneva commented:
+tests for `np.round` and `DataFrame.round` to ensure that the column names are preserved and that the rounding isn't otherwise affected.

jreback commented on pandas/tests/test_frame.py:
```diff
@@ -13542,6 +13542,14 @@ def test_round(self):
                 df.round(nan_round_Series)
 
+        # named columns
+        df2 = df.copy()
```
add the issue number here as a comment

jreback commented on pandas/tests/test_frame.py:
```diff
@@ -13542,6 +13542,14 @@ def test_round(self):
+        # named columns
+        df2 = df.copy()
+        df2.columns.name = "cols"
+        tm.assert_equal(df2.round().columns.name, "cols")
```
instead of doing this, just add the names onto the ``expected_rounded`` guy (and the original df). the ``assert_frame_equal`` automatically checks for names (since they are both ``None`` it didn't make a difference before though)

rcarneva commented:
Better like this?

jreback commented:
yes although I wouldn't change it back to None (just change any expected after this)

jreback commented:
can u update the series test as well (add a name)
it should work already

rcarneva commented:
I'd put the new test in the middle of the rounding tests, so there would have been a lot of expected frames to fix without resetting the name. I've moved it to the bottom now, and added tests to ensure that naming the columns doesn't mess up `Series.round` or the name of a selected `Series` after a `DataFrame.round`.

jreback commented on pandas/tests/test_frame.py:
```diff
@@ -13545,6 +13545,18 @@ def test_round(self):
+        expected_rounded.columns.name = "cols"
+        tm.assert_frame_equal(df.round(decimals), expected_rounded)
+
+        # interaction of named columns & series
```
this last needs to go in test_series where we test round

rcarneva replied:
I think that would end up duplicating the existing `Series.round` tests since this change only affects the name of the index representing the columns, not the column names themselves.

jreback replied:
this is trivial, just add a name to the index in the Series tests for .round.

rcarneva replied:
Gotcha, thanks - added it. As you mentioned, that behavior was already working and wouldn't have been affected by the change, so it just wasn't in my brain.

jreback commented:
pls add an entry in whatsnew for this bug fix.

jreback commented:
gr8. ping when green.

rcarneva commented:
Green. Thanks for the help with this, @jreback

jreback commented:
merged via https://github.com/pydata/pandas/commit/9b5c052b0807e4990dd5a8fe39238153c2ec9632

thanks!
